MockerTests/MockerTests.swift

```diff
@@ -401,4 +408,126 @@ final class MockerTests: XCTestCase {
+            expectation.fulfill()
+        }.resume()
+        
+        wait(for: [expectation], timeout: 2.0)
+    }
+    
+    /// `onMockFor` returning nil for non-matching replacement mock shouldn't change previous behavior
+    func testOnMockForReplacementReturnsNil() {
+        let url = URL(string: "https://www.fakeurl.com")!
+        let mockData = "{\"key\":\"value\"}".data(using: .utf8)!
+        
+        Mocker.onMockFor = { (request: URLRequest) -> Mock? in
+            return nil
+        }
+        
+        let expectation = self.expectation(description: "Data request should succeed")
+        
+        let mock = Mock(url: url, dataType: .json, statusCode: 200, data: [.get: mockData])
+        mock.register()
+        
+        URLSession.shared.dataTask(with: URLRequest(url: url)) { (data, urlresponse, err) in
+            
+            XCTAssertEqual(data, mockData, "Data from mock not present")
+            XCTAssertNotNil(urlresponse)
+            XCTAssertNil(err)
+            
+            expectation.fulfill()
+        }.resume()
+        
+        wait(for: [expectation], timeout: 2.0)
+    }
+    
+    /// `onMockFor` can change the Mock used for a url for subsequent fetches during same test
+    func testOnMockForConditionalReplacement() {
+        var problemFixed = false    // imaginary problem flag
+        
+        let url = URL(string: "https://www.fakeurl.com/url1")!
+        let goodMockData = "good".data(using: .utf8)!
+        
+        let failingMock = Mock(url: url, dataType: .json, statusCode: 401, data: [.get: Data()])
+        failingMock.register()
+        
+        let succeedingMock = Mock(url: url, dataType: .json, statusCode: 200, data: [.get: goodMockData])
+        // succeedingMock.register() - Do NOT call now as will replace failingMock.
+        // Instead, return `succeedingMock` using new `onMockFor` handler if problem is fixed:
+        
+        Mocker.onMockFor = { _ in
+            if problemFixed {
+                return succeedingMock
+            } else { return nil }
+        }
+        
+        let expectFailure = expectation(description: "call url - should get error")
+        
+        URLSession.shared.dataTask(with: URLRequest(url: url)) { (data, urlresponse, err) in
+            XCTAssertNotNil(data)
+            XCTAssert((urlresponse as? HTTPURLResponse)?.statusCode == 401)
+            XCTAssertNil(err)
+            
+            problemFixed = true         // Imagine problem was handled internally by library that
+                                        // then retries automatically (next call to same url)
+            expectFailure.fulfill()
+        }.resume()
+        
+        // Wait to call second "retry" fetch until first one returns.
+        wait(for: [expectFailure], timeout: 2.0)
+        
+        let expectSuccess = expectation(description: "call url again - should succeed")
+        
+        // This retry should succeed becuase problemFixed is now true
```

Comment:
```suggestion
        // This retry should succeed because `problemFixed` is now `true`
```

Reply:
I'm happy to make this change but thought I'd mention that the reason I don't generally use backticks in internal quotes is because they are not displayed in Quick Help and so are only // style comments where backticks don't create a style change anywhere.

I'm curious if there's a place internal quotes show up for which the styling elements are thus useful because if so I'll need to change my conventions.

Reply:
I guess it's useful to clarify when the comment is referring to symbols or not even if it's not displayed differently.

Reply:
I guess if you make use of three slashes it should work `///`

Reply:
I do not see that comments within the body of a function are pulled into the Quick Help for the function (Xcode 11.5).  If they appear somewhere else then I'd like to know about that.

In discussion with my colleagues we decided that even if it doesn't show up it helps the reader know you are referring to a symbol.

Reply:
Ah, I didn't see that this was an inline comment. Indeed, those are not showing in Quick Help but I agree that it still helps the reader!